File: java-api-injection/src/main/java/Java_Api_Injection/service/UserService.java
package Java_Api_Injection.service;

import java.util.List;

import org.springframework.stereotype.Service;

import Java_Api_Injection.constant.InjectionConstant;
import Java_Api_Injection.dao.UserDao;
import Java_Api_Injection.model.User;

@Service
public class UserService {
  private final boolean SQL_INJECTION_ACTIVE = InjectionConstant.SQL_INJECTION_ACTIVE;
  UserDao userDao = new UserDao();

  public List<User> getAll() {
    return userDao.getAll();
  }

  public User get(int id) {
    return SQL_INJECTION_ACTIVE ? userDao.get(id) : userDao.getSecure(id);
  }

  public List<User> search(String input) {
    return SQL_INJECTION_ACTIVE ? userDao.search(input) : userDao.searchSecure(input);
  }

  public User login(User user) {
    return SQL_INJECTION_ACTIVE ? userDao.login(user) : userDao.loginSecure(user);
  }
}

File: node-api-injection/api/src/services/UserService.js
const { SQL_INJECTION_ACTIVE } = require('../constants/InjectionConstant.js')
const userModel = require('../models/UserModel.js')

const getAll = async () => {
  return await userModel.getAll()
}

const get = async (id) => {
  return SQL_INJECTION_ACTIVE
    ? await userModel.get(id)
    : userModel.getSecure(id)
}

const search = async (input) => {
  return SQL_INJECTION_ACTIVE
    ? await userModel.search(input)
    : userModel.searchSecure(input)
}

const login = async (userLogin) => {
  return SQL_INJECTION_ACTIVE
    ? await userModel.login(userLogin)
    : userModel.loginSecure(userLogin)
}

module.exports = {
  getAll,
  get,
  search,
  login,
}
